include/swift/Runtime/CustomRRABI.h

```diff
@@ -0,0 +1,90 @@
+//===--- CustomRRABI.h - Custom retain/release ABI support ----------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Utilities for creating register-specific retain/release entrypoints.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_RUNTIME_CUSTOMRRABI_H
+#define SWIFT_RUNTIME_CUSTOMRRABI_H
+
+namespace swift {
+
+#if __arm64__
```

Comment:
Please add in `defined(_M_ARM64)`.

Reply:
If we're doing that, we should presumably also add in `x18` (I'm assuming it isn't reserved on Windows?)

Reply:
That also begs the question of whether we're doing this on Windows _at all_. If we are, we should check how Windows uses the registers — the Windows ABI is very likely different to the System V ABI that gets used elsewhere.

Reply:
Surprisingly (at least to me), it looks like Windows is the same in this respect. x18 is reserved (here, it's apparently used to point to data for the current thread), and x16/x17 are scratch registers for procedure calls. https://learn.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions?view=msvc-170

Reply:
:-) Well that simplifies things.

Reply:
It's nice to have everything on the same page for once. I added `|| defined(_M_ARM64)` so this will be available on Windows as well.

Reply:
Don't worry, LP64 vs LLP64 will still be there to defeat us @mikeash :-p

Reply:
That's a relief.